python-pyo3/python/n_vault/vault.py

```diff
@@ -112,6 +118,8 @@ def delete_many(self, names: Collection[str]) -> None:
             key=self.vault_key,
             prefix=self.vault_prefix,
             profile=self.profile,
+            iam_id=self.vault_iam_id,
```

Comment:
I wonder if there would some better method passing these values as some combined struct/class but might not be🤷. Seems quite verbose😅. This is just an observation, nothing else.

Reply:
Yes, I was thinking of making a `VaultConfig` etc dataclass that would contain the settings but left it like this so the interface stays the same as before. Could add a separate factory method for the rust vault that takes a single config object, will try it out later today.